### ISSUE TITLE
no suitable peers available

### ISSUE THREAD

**Issue by slowkow** [golemfactory/clay#1001]
System: Ubuntu 17.04, 1GB ram

I installed golemapp as follows:

```bash
wget https://raw.githubusercontent.com/golemfactory/golem/develop/Installer/Installer_Linux/install.sh
chmod +x install.sh
./install.sh
sudo apt install docker
sudo apt instal scrypt
pip install scrypt
```

I ran it like this:

```
golemapp --nogui
```

After some time, I get an error repeatedly (snippet shown below).

Then I tried restarting the app, and the error stopped appearing.

## Error

```
INFO [04-17|14:10:32] Imported new block headers               count=192  elapsed=95.687ms  number=766080 hash=d189a8…eb82e8 ignored=0
INFO [04-17|14:10:32] Imported new block headers               count=192  elapsed=119.900ms number=766272 hash=fcf8db…391807 ignored=0
INFO [04-17|14:10:35] Imported new block headers               count=192  elapsed=3.396s    number=766464 hash=bbcd3a…28f973 ignored=0
INFO [04-17|14:10:36] Imported new block headers               count=28   elapsed=479.638ms number=766492 hash=dff2c1…198407 ignored=0
ERROR    [golem.pay                          ] IPC error: {u'message': u'no suitable peers available', u'code': -32000}
INFO     [golem.client                       ] Block synchronization complete
INFO     [golem.client                       ] Starting network ...
HyperG is ready
WARNING  [golem.docker.manager               ] Docker machine is not available: native Linux environment
WARNING  [golem.docker.manager               ] Docker: pulling image golemfactory/base:1.2
INFO [04-17|14:10:53] Imported new block headers               count=3    elapsed=64.752ms  number=766495 hash=ce53a7…be376d ignored=0
WARNING  [golem.docker.manager               ] Docker: pulling image golemfactory/blender:1.3
INFO [04-17|14:11:05] Imported new block headers               count=2    elapsed=45.536ms  number=766497 hash=3f2c47…79c142 ignored=0
WARNING  [golem.docker.manager               ] Docker: pulling image golemfactory/luxrender:1.2
ERROR    [golem.network.p2p.p2pservice       ] Invalid seed address: Empty host name
ERROR    [golem.network.p2p.p2pservice       ] Invalid seed address: Empty host name
INFO     [golem.client                       ] Starting p2p server ...
INFO     [golem.client                       ] Starting resource server...
INFO     [golem.client                       ] Starting task server ...
INFO     [golem.rpc.crossbar                 ] New node key pair generated!
INFO     [golem.rpc.crossbar                 ] File permissions on node public key fixed!
INFO     [golem.rpc.crossbar                 ] File permissions on node private key fixed!
INFO     [golem.rpc.crossbar                 ] No extra node router roles
INFO     [golem.rpc.crossbar                 ] Router service session attached [crossbar.router.service.RouterServiceSession]
INFO     [golem.rpc.crossbar                 ] Node controller attached [crossbar.controller.process.NodeControllerSession]
INFO     [golem.rpc.crossbar                 ] Using default node shutdown triggers [u'shutdown_on_worker_exit']
INFO     [golem.rpc.crossbar                 ] Configuring node from local configuration ...
INFO     [golem.rpc.crossbar                 ] Starting 1 workers ...
ERROR    [golem                              ] Service Error: Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_threadworker.py", line 46, in work

  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_team.py", line 190, in doWork

--- <exception caught here> ---
  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 250, in inContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 122, in callWithContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 85, in callWithContext

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 310, in _run

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 106, in synchronized

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 97, in check

  File "/home/mfrancis/Projects/golem/golem/ethereum/client.py", line 53, in is_syncing

  File "/usr/local/lib/python2.7/dist-packages/web3/utils/functional.py", line 14, in inner

  File "/usr/local/lib/python2.7/dist-packages/web3/eth.py", line 163, in getBlock

  File "/usr/local/lib/python2.7/dist-packages/web3/providers/manager.py", line 51, in request_blocking

exceptions.ValueError: {u'message': u'no suitable peers available', u'code': -32000}

None
INFO     [golem.rpc.crossbar                 ] Router 'worker-001': realm 'realm-001' (named 'golem') started
INFO     [golem.rpc.crossbar                 ] Router 'worker-001': role 'role-001' (named 'anonymous') started on realm 'realm-001'
INFO     [golem.rpc.crossbar                 ] Router 'worker-001': transport 'transport-001' started
INFO     [golem.rpc.crossbar                 ] Node configuration applied successfully!
INFO [04-17|14:11:48] Imported new block headers               count=2    elapsed=48.992ms  number=766499 hash=88f053…ba2da3 ignored=0
INFO [04-17|14:11:58] Imported new block headers               count=2    elapsed=17.376ms  number=766501 hash=5c05ce…ff00b7 ignored=0
INFO [04-17|14:12:10] Imported new block headers               count=2    elapsed=17.185ms  number=766503 hash=b0b4ea…7f6a69 ignored=0
ERROR    [golem                              ] Service Error: Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_threadworker.py", line 46, in work

  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_team.py", line 190, in doWork

--- <exception caught here> ---
  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 250, in inContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 122, in callWithContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 85, in callWithContext

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 310, in _run

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 106, in synchronized

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 97, in check

  File "/home/mfrancis/Projects/golem/golem/ethereum/client.py", line 53, in is_syncing

  File "/usr/local/lib/python2.7/dist-packages/web3/utils/functional.py", line 14, in inner

  File "/usr/local/lib/python2.7/dist-packages/web3/eth.py", line 163, in getBlock

  File "/usr/local/lib/python2.7/dist-packages/web3/providers/manager.py", line 51, in request_blocking

exceptions.ValueError: {u'message': u'no suitable peers available', u'code': -32000}

None
ERROR    [golem                              ] Service Error: Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_threadworker.py", line 46, in work

  File "/usr/local/lib/python2.7/dist-packages/twisted/_threads/_team.py", line 190, in doWork

--- <exception caught here> ---
  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 250, in inContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 122, in callWithContext

  File "/usr/local/lib/python2.7/dist-packages/twisted/python/context.py", line 85, in callWithContext

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 310, in _run

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 106, in synchronized

  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 97, in check

  File "/home/mfrancis/Projects/golem/golem/ethereum/client.py", line 53, in is_syncing

  File "/usr/local/lib/python2.7/dist-packages/web3/utils/functional.py", line 14, in inner

  File "/usr/local/lib/python2.7/dist-packages/web3/eth.py", line 163, in getBlock

  File "/usr/local/lib/python2.7/dist-packages/web3/providers/manager.py", line 51, in request_blocking

exceptions.ValueError: {u'message': u'no suitable peers available', u'code': -32000}

None
```

**Comment by josephxsxn** [comment]
Try - 
`golemapp --nogui --peer 86.3.196.111:40102 --peer 76.250.215.80:40102
`
Then you can validate it by 
```
golemcli -i
>> network connect $YOUR_PUB_IP 40102
Completed in 0.00135707855225s
>>network status
Connected
>>network show
ip               port  id                                   name
-------------  ------  -----------------------------------  ------------------
86.3.196.111    53567  85f0b31763588e69...75c0cd9bf2e9325c  michal.brazewicz.2
170.84.135.59   12784  b59de173ee9baf11...527feeeac63ab724  nicanode
194.181.80.71   40112  3601923d6ff99410...6b4589ea8cff2205  Blablaryka5
94.23.196.166   52930  86b2f6c15d20f363...23866cc72a9a8691  drohlum/3
```

**Comment by badb** [comment]
Solution proposed by jesephxsxn won't help because that is the problem with ethereum nodes seeds, not Golem newtork seeds. 
We may want to catch this type of errors and show user a proper warning / information about problem. Apart from need for better error handling, this should be a temporary issue caused by problem with ethereum testnet. 
@slowkow Can you confirm that you don't have this problem any more?

**Comment by slowkow** [comment]
With v0.5.3 this error has not appeared after running for 10 minutes.

**Comment by anthonyms** [comment]
Hopefully, this is the right place to report this. I'm getting an error very similar to the original error.

```
ERROR    [golem                              ] Service Error: Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/_threads/_threadworker.py", line 46, in work
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/_threads/_team.py", line 190, in doWork
    
--- <exception caught here> ---
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    
  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 310, in _run
    
  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 106, in synchronized
    
  File "/home/mfrancis/Projects/golem/golem/ethereum/paymentprocessor.py", line 93, in check
    
  File "/home/mfrancis/Projects/golem/golem/ethereum/client.py", line 42, in get_peer_count
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/web3/utils/functional.py", line 14, in inner
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/web3/net.py", line 24, in peerCount
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/web3/providers/manager.py", line 43, in request_blocking
    
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/web3/providers/ipc.py", line 89, in make_request
    
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/home/mfrancis/Projects/golem/venv/lib/python2.7/site-packages/web3/providers/ipc.py", line 37, in get_ipc_socket
    
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 111] Connection refused
```

I'm Ubuntu, Golem v0.5.3, installed via the install.sh, running the command by @josephxsxn,
Output of the validation is as below


```
sudo ./golemcli -i
>> 
>> network connect 139.59.123.190 40102
Completed in 0.0129680633545s
>> network status
Connected
>> network show
ip               port  id                                   name
-------------  ------  -----------------------------------  ------------
194.181.80.71   40112  3601923d6ff99410...6b4589ea8cff2205  Blablaryka5
94.23.196.166   40102  86b2f6c15d20f363...23866cc72a9a8691  drohlum/3
2.50.50.98      61242  ed879cbe9c6499a1...46935f7ff1cafb7d  pktest
194.181.80.91   35809  545a6438330be11e...72d3bf3630af0df6  chfast-imapp
99.27.120.250   57387  5abcd84baef5650c...6bd0f273cedd9959  bgb
>> exit
Completed in 0.00132822990417s
```

**Comment by paulperegud** [comment]
@slowkow as a temporary workaround, start geth separately from golem with following parameters:
`geth --testnet --fast`
This will start geth in full node mode so it will not need remote node with free sockets for light clients. Please also node that ropsten testnet is very unreliable now because of attacks and general lack of reliability of PoW testnets.

**Comment by anthonyms** [comment]
Thanks, @paulperegud this also solved it for me.  Or rather helped me figure out the problem. `geth` was failing because of low RAM.

I added more RAM, started `geth` and the problem disappeared.

Do we have documentation for recommended hardware specs?

**Comment by andrewnguyen22** [comment]
I am having a similar issue with v 0.6.0
@paulperegud & @josephxsxn solutions did not solve my issue. 
If anyone can help please do

```
Failed to import scrypt. This is not a fatal error but does
mean that you cannot create or decrypt privkey jsons that use
scrypt

WARNING  [golem.appconfig                    ] Can't open file /home/manager/.local/share/golem/SimpleEnv/minilight.ini: [Errno 2] No such file or directory: '/home/manager/.local/share/golem/SimpleEnv/minilight.ini'
INFO     [golem.client                       ] Client "", datadir: /home/manager/.local/share/golem/default 
INFO     [golem.client                       ] Starting network ... 
WARNING  [golem.docker.manager               ] Docker machine is not available: native Linux environment
INFO     [golem.client                       ] Starting resource server ... 
CRITICAL [golem.resources                    ] Can't run hyperdrive executable 'hyperg --db /home/manager/.local/share/golem/default/hyperg'. Make sure path is correct and check if it starts correctly.
WARNING  [golem.docker.task_thread           ] Task stderr:
['blender', '-b', '/golem/resources/scene-Helicopter-27-cycles.blend', '-y', '-P', '/golem/work/blenderscript.py', '-o', '/golem/output/blender_benchmark_1', '-F', 'PNG', '-t', '4', '-f', '1']
Traceback (most recent call last):
  File "/golem/work/job.py", line 57, in <module>
    params.output_format)
  File "/golem/work/job.py", line 51, in run_blender_task
    exit_code = exec_cmd(cmd)
  File "/golem/work/job.py", line 16, in exec_cmd
    pc = subprocess.Popen(cmd)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

ERROR    [golem.task.taskthread              ] Task computing error: Subtask computation failed with exit code 1
WARNING  [golem.task                         ] Failed to compute benchmark Subtask computation failed with exit code 1
ERROR    [golem.task.taskcomputer            ] Unable to run lux benchmark: Subtask computation failed with exit code 1
ERROR    [golem.network.p2p.p2pservice       ] Invalid seed address: Empty host name
```

**Comment by badb** [comment]
- We moved to rinkeby, so geth testnet issues should be solved.
- Non-critical geth errors are caught. 
- Benchmark bug mentioned by aesopwonder was  resolved by #1171.
